api/proto/banyandb/database/v1/rpc.proto

```diff
@@ -354,10 +355,56 @@ message GroupRegistryServiceUpdateResponse {}
 
 message GroupRegistryServiceDeleteRequest {
   string group = 1;
+  bool dry_run = 2;
+  bool force = 3;
 }
 
 message GroupRegistryServiceDeleteResponse {
   bool deleted = 1;
+  string task_id = 2;
+}
+
+message GroupDeletionProgress {
+  enum Phase {
+    PHASE_UNSPECIFIED = 0;
+    PHASE_PENDING = 1;
+    PHASE_IN_PROGRESS = 2;
+    PHASE_COMPLETED = 3;
+    PHASE_FAILED = 4;
+  }
+  Phase current_phase = 1;
+  int32 stream_count = 2;
+  int32 measure_count = 3;
+  int32 trace_count = 4;
```

Comment:
Lack of the property data module. Or you can use a "map" instead of dedicated fields.